header-rewrite-filter/header_processor.cc

```diff
@@ -149,8 +149,20 @@ namespace HeaderRewriteFilter {
         // cast to RequestHeaderMap because setPath is only on request side
         Http::RequestHeaderMap* request_headers = static_cast<Http::RequestHeaderMap*>(&headers);
         
-        // set path (path being set here includes the query string)
-        request_headers->setPath(request_path); // should never return an error
+        // get path (includes query string)
+        absl::string_view path = request_headers->path();
+
+        const size_t offset = path.find_first_of("?"); // TODO: envoy implements this as ?#
```

Comment:
Could you use https://github.com/envoyproxy/envoy/blob/main/source/extensions/http/header_validators/envoy_default/path_normalizer.cc#L427 instead?

Reply:
It looks like the [visibility for this function](https://github.com/envoyproxy/envoy/blob/main/source/extensions/http/header_validators/envoy_default/BUILD#L48-L57) is not public

Reply:
ack